### DISCUSSION TITLE
Firefox bug related to 'delete cookies... on exit' causes client-side JS to fail

### DISCUSSION THREAD

**Discussion by zEdS15B3GCwq** [coderaiser/cloudcmd#343] [Q&A] [answered]
Hi,

I've set up Cloud Commander on a remote box and accessing it through an SSH tunnel - this works perfectly. However, when I proxy pass to CC via Apache2, although the main page loads, none of the JS actions seem to be working - e.g. the right-click menu is unchanged, and the bottom menu bar items don't do anything. What am I doing wrong?

Any help would be greatly appreciated.

Both apache2 and cc are in docker containers, running on a common bridge network.

My setup in httpd.conf:

```
    <Location /cloudcmd/>
        AuthType basic
        AuthName "cloudcmd"
        AuthUserFile /usr/local/apache2/secrets/htpasswd
        <RequireAll>
            Require user cloudcmd
            Require ssl
        </RequireAll>
        ProxyPass http://cloudcmd:8000/cloudcmd/
        ProxyPassReverse http://cloudcmd:8000/cloudcmd/
        ProxyPassReverseCookieDomain localhost xxx.yyy.zzz.www
        ProxyPassReverseCookiePath / /cloudcmd
    </Location>

```
.cloudcmd.json:

```
    "name": "",
    "auth": false,
    "username": "",
    "password": "",
    "algo": "sha512WithRSAEncryption",
    "editor": "edward",
    "packer": "tar",
    "diff": true,
    "zip": true,
    "buffer": true,
    "dirStorage": false,
    "online": false,
    "open": false,
    "keysPanel": true,
    "port": 8000,
    "ip": null,
    "root": "/mnt/fs",
    "prefix": "/cloudcmd",
   ...
```

cc in compose.yml:

```
  cloudcmd:
    image: coderaiser/cloudcmd:latest-alpine
    container_name: cloudcmd
    networks:
      - front
    ...
    ports:
      - 127.0.0.1:35985:8000

```

CC accessed on localhost:8000 works as expected, accessing it through apache2 causes JS to not work (ports published or not).

**Comment by coderaiser** [comment]
Could you please tell me, do you have any errors in console of your browser or network panel?

**Reply by zEdS15B3GCwq** [reply]
Hi, thanks for getting back to me so quickly.

This is what I see in FF Console after initially loading CC:

```
Failed to register/update a ServiceWorker for scope ‘https://MY_IP:MY_PORT/cloudcmd/’: Storage access is restricted in this context due to user settings or private browsing mode. cloudcmd.js:1:1791

Content Security Policy: The report URI (about:blank) should be an HTTP or HTTPS URI. cloudcmd.js:1:1791

Content Security Policy: The page’s settings observed the loading of a resource at https://MY_IP:MY_PORT/cloudcmd/sw.js (“worker-src”). A CSP report is being sent. cloudcmd.js:1:1791

Requesting search results.. search_results.js:1:9

Uncaught (in promise) DOMException: The operation is insecure. cloudcmd.js:1
    CloudCmd https://MY_IP:MY_PORT/cloudcmd/dist/cloudcmd.js:1
    AsyncFunctionThrow self-hosted:694
    (Async: async)
    <anonymous> https://MY_IP:MY_PORT/cloudcmd/fs/:282

Source map error: Error: request failed with status 401
Resource URL: https://MY_IP:MY_PORT/cloudcmd/dist/cloudcmd.js
Source Map URL: cloudcmd.js.map
```

**Reply by coderaiser** [reply]
Looks like you using incognito mode, is it `chrome`? Can you reproduce this in regular mode?

**Reply by zEdS15B3GCwq** [reply]
Firefox regular mode, not incognito. Direct access through HTTP and proxied through Apache/HTTPS, these 2 tabs are open in the same window, affected by the same plugins and config (I assume). I'll do some testing tomorrow to see if disabling extensions makes a difference. It's also possible that Firefox applies stricter limitations when on HTTPS?

**Reply by zEdS15B3GCwq** [reply]
Seems it's a FF bug:

https://bugzilla.mozilla.org/show_bug.cgi?id=1413615
https://github.com/mozilla/send/issues/1222

Unchecking "delete cookies and site data when FF is closed" eliminates the first error and JS actions actually seem to work now - context menu and menu bar buttons. Creating an exception for the site works as well.

**Reply by coderaiser** [reply]
So you still see an exception?

```
Uncaught (in promise) DOMException: The operation is insecure. cloudcmd.js:1
    CloudCmd https://MY_IP:MY_PORT/cloudcmd/dist/cloudcmd.js:1
    AsyncFunctionThrow self-hosted:694
    (Async: async)
    <anonymous> https://MY_IP:MY_PORT/cloudcmd/fs/:282
```

Could you please try if [cloudcmd demo](https://cloudcmd.herokuapp.com) works for you?

**Reply by zEdS15B3GCwq** [reply]
No, sorry, that exception doesn't show if the "delete cookies..." option is off.

All I can see now is:

```
Content Security Policy: The report URI (about:blank) should be an HTTP or HTTPS URI. register.js:20:41

Content Security Policy: The page’s settings observed the loading of a resource at https://MY_IP:MY_PORT/cloudcmd/sw.js (“worker-src”). A CSP report is being sent. register.js:20:41

```

I'll try the demo soon.

**Reply by zEdS15B3GCwq** [reply]
demo - same situation, same errors as with the https proxied version. Turning the 'delete cookies' setting off (or adding the site as an exception) brings back the lost functionality.

**Reply by coderaiser** [reply]
Looks like a bug in Firefox, because `Cloud Commander` doesn’t use cookies.

**Reply by zEdS15B3GCwq** [reply]
Yes, based on the bug reports, it certainly is. Thanks for giving me hints to solve this issue.

**Comment by zEdS15B3GCwq** [comment, marked as answer]
Answer summarised:

This is a FF bug, discussed here:

https://bugzilla.mozilla.org/show_bug.cgi?id=1413615
https://github.com/mozilla/send/issues/1222

Possible solution(s):

1. Unchecking "delete cookies and site data when FF is closed", or
2. Adding an exception for the site in the same option
